edgexfoundry/device-usb-camera
# feat: select stream for api commands

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have added unit tests for the new feature or bug fix (if not, why?) I've only added device level functionality
  <link to docs PR>

## Testing Instructions

1. Execute each command with no query parameters and see status code 200.
2. Execute each command with the `PathIndex` set to the appropriate value. For standard usb cameras, this can only be 0, and for RealSense cameras this can be 0, 1, or 2. If the index is outside of that range, it will send an error.
3. Execute each command with the `StreamFormat` set to the appropriate value. For standard usb cameras, this can only be "RGB", and for RealSense cameras this can be "RGB", "Greyscale" or "Depth". If the value is not supported, it will send an error.
4. For realsense, start streaming for each index and check that the stream is working (using vlc or mplayer) and there are no errors. Also make sure that the streams are accurate, ie rgb shows rgb, etc. Make sure to stop streaming after checking the stream.
Note: Some commands will return the "/dev/video" stream. Make sure that matches the stream that you supplied in the query parameter.
Note: The stop streaming command does not need one of these query parameters.

## New Dependency Instructions (If applicable)
